src/menu/wallet_menu.c

```diff
@@ -0,0 +1,201 @@
+ * @param data_ptr Currently unused pointer set by the engine
+ */
+static void ignore_p0_handler(engine_ctx_t *ctx,
+                              p0_evt_t p0_evt,
+                              const void *data_ptr);
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+static uint8_t wallet_index = 0xFF;
+
+static const flow_step_t wallet_step = {.step_init_cb = wallet_menu_initialize,
+                                        .p0_cb = ignore_p0_handler,
+                                        .ui_cb = wallet_menu_handler,
+                                        .usb_cb = NULL,
+                                        .nfc_cb = NULL,
+                                        .evt_cfg_ptr = &device_nav_evt_config,
+                                        .flow_data_ptr = &wallet_index};
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+static void ignore_p0_handler(engine_ctx_t *ctx,
+                              p0_evt_t p0_evt,
+                              const void *data_ptr) {
+  ignore_p0_event();
+}
+
+static void wallet_menu_initialize(engine_ctx_t *ctx, const void *data_ptr) {
+  char *wallet_list[MAX_WALLETS_ALLOWED] = {0};
+  get_wallet_list((const char **)wallet_list);
+
+  menu_init((const char **)ui_text_options_old_wallet,
+            NUMBER_OF_OPTIONS_OLD_WALLET,
+            (const char *)wallet_list[wallet_index],
+            true);
+  return;
+}
+
+static void wallet_menu_handler(engine_ctx_t *ctx,
+                                ui_event_t ui_event,
+                                const void *data_ptr) {
+  if (UI_EVENT_LIST_CHOICE == ui_event.event_type) {
+    switch (ui_event.list_selection) {
+      case VIEW_SEED: {
+        // TODO: Handle view seed flow
+        break;
+      }
+      case DELETE_WALLET: {
+        // TODO: Handle delete wallet flow
+        break;
+      }
+      default: {
+        break;
+      }
+    }
+  } else {
+    // UI_EVENT_LIST_REJECTION handled below already
+  }
+
+  /* Return to the previous menu irrespective if UI_EVENT_REJECTION was
+   * detected, or a wallet flow was executed */
+  engine_delete_current_flow_step(ctx);
+
+  return;
+}
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+const flow_step_t *wallet_menu_get_step(uint8_t selection_index) {
+  if (0 == selection_index || MAX_WALLETS_ALLOWED < selection_index) {
+    return NULL;
+  }
```

Comment:
How will this be caught?

Reply:
Agreed we don't want this case to occur at all.

Reply:
updated